src/gui/socketapi.cpp

```diff
@@ -763,12 +798,24 @@ void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListe
     if (folder) {
         auto virtualFileSuffix = QStringLiteral(APPLICATION_DOTVIRTUALFILE_SUFFIX);
         bool hasVirtualFile = false;
+        bool hasNormalFiles = false;
+        bool hasDir = false;
         for (const auto &file : files) {
-            if (file.endsWith(virtualFileSuffix) || (folder->useVirtualFiles() && QFileInfo(file).isDir()))
+            if (QFileInfo(file).isDir()) {
+                hasDir = true;
+            } else if (file.endsWith(virtualFileSuffix)) {
                 hasVirtualFile = true;
+            } else if (!hasNormalFiles) {
+                FileData fileData = FileData::get(files.first());
+                bool isOnTheServer = fileData.journalRecord().isValid();
+                hasNormalFiles = isOnTheServer;
+            }
         }
-        if (hasVirtualFile)
+        if (hasVirtualFile || (hasDir && folder->useVirtualFiles()))
             listener->sendMessage(QLatin1String("MENU_ITEM:DOWNLOAD_VIRTUAL_FILE::") + tr("Download file(s)", "", files.size()));
+
+        if ((hasNormalFiles || hasDir) && folder->useVirtualFiles())
+            listener->sendMessage(QLatin1String("MENU_ITEM:REPLACE_VIRTUAL_FILE::") + tr("Replace file(s) by virtual file", "", files.size()));
```

Comment:
The wording isn't as immediately clear as "Download file(s)". 

"Remove local copy of file contents"
"Free space by dropping file contents"
"Store file(s) on the server only"

Unsure.

Reply:
@michaelstingl   : What  should the wording be?